### PR TITLE
#213: Add getRequired() method to the Gettable interface.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java [hunk 1]
```diff
@@ -23,4 +25,11 @@
    */
   T get();
 
+
+  /**
+   * @return the item or throws Exception if the item doesn't exist.
+   * @throws io.fabric8.kubernetes.client.KubernetesClientException if an error occurs
+   */
+  T getRequired() throws ResourceNotFoundException;
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java]: How about `mustGet`?

Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java]: TBH, I don't like mustGet, I also don't like getRequired(), getMandatory() or even require().

I was tempted to make it part of the DSL as an additional method like: .failingIfNotExists().get(); But that also seems like an overkill. So not sure ....

Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java]: made up my mind..... it should be getOrBarf() :-)

Comment (jstrachan) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java]: getNotNull()? Guava has [checkNotNull()](http://guava-libraries.googlecode.com/svn-history/r14/trunk/javadoc/com/google/common/base/Preconditions.html#checkNotNull(T)) though thats more of a precondition check. how about getAndNotNull()?

Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java]: Let's summarise our options:

i) require()
ii) xxx.getRequired()
iii) xxx.getMandatory()
iv) xxx.mustGet()
v) xxx.getAndNotNull()
vi) xxx.failingIfMissing().get();

**Comment by jimmidyson** [comment]
What's happening with this?

**Comment by iocanel** [comment]
Once, I find some spare time, I I'll go with @davsclaus suggestion, which is much cleaner.

**Comment by iocanel** [comment]
almost a year old, lets drop it

**Pull request closed without merging**